src/basic/path-util.c

```diff
@@ -637,6 +637,9 @@ int find_binary(const char *name, char **ret) {
                 if (!j)
                         return -ENOMEM;
 
+                if (is_dir(j, true))
```

Comment:
Should we also check for `is_dir()` on the block above, lines 601-612?

Otherwise, looks great!

Reply:
That one only passes is `is_path`, which is `!!strchr(p, '/')`, so in that case I think we probably *do* want to show this error :-)